deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: gitsources.devopsconsole.openshift.io
+spec:
+  group: devopsconsole.openshift.io
+  names:
+    kind: GitSource
+    listKind: GitSourceList
+    plural: gitsources
+    singular: gitsource
+    shortNames:
+      - gitsrc
+  scope: Namespaced
+  validation:
+    # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            url:
+              type: string
+            ref:
+              type: string
+            contextDir:
+              type: string
+            httpProxy:
+              type: string
+            httpsProxy:
+              type: string
+            noProxy:
+              type: string
+            sourceSecret:
+              type: string
```

Comment:
I would probably store secrets separated and have `sourceSecret` as a reference to this CR (eg. `GitSecret` or in general `SourceSecret`). The reason is that the secret is usually associated more with a user account than with a repo. If we had `GitSecret` separated it would make changes/updates of the secret much easier across all `GitSources`. 
Imagine that one user has multiple `GitSource` CRs for several repositories/branches. This user has one GH token that uses for all of the `GitSources` and after some time he would need to change the token. To be honest, I'm not sure how we would manage such a change across all `GitSources` if we didn't have the secret stored seperated.

Reply:
Yes, this should have the name of the `sourceSecret`. Are you suggesting we should have the `type` as `kind` `secret` ?

Reply:
Either secret or more likely a new custom resource. For example something like this:
```yaml
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
...
spec:
  group: devopsconsole.openshift.io
  names:
    kind: SourceSecret
    ...
  validation:
    openAPIV3Schema:
      properties:
        apiVersion:
          type: string
        kind:
          type: string
        metadata:
          type: object
        spec:
          properties:
            flavor:
              type: string
            secret:
              type: secret
```

Reply:
`sourceSecret` is not the credentials itself. It's just a name of the secret in the same namespace. So, basically it's already a reference to a secret.
Maybe `sourceSecret` is not the best name for this field because it can confuse. Will change it to something more clear.
The secret itself will keep different type of Git credentials. SSH key, token (for github for example), certificate, etc. Currently OpenShift build already can handle it and use the secret properly and I think we should leverage it.

Reply:
What about `secretRef` as the name?

Reply:
Yeah.. `secretRef` looks good to me.

Reply:
Changed it to
```
  secretRef:
    name: mysecret
```

Reply:
Ok, I misundestood that based of the name and its original value (only string). Thanks for changing it @alexeykazakov, it looks much better now